src/dev/register_git_hook/register_git_hook.js

```diff
@@ -22,13 +22,32 @@ import { chmod, unlink, writeFile } from 'fs';
 import dedent from 'dedent';
 import { resolve } from 'path';
 import { promisify } from 'util';
+import SimpleGit from 'simple-git';
 import { REPO_ROOT } from '../constants';
 
+const simpleGit = new SimpleGit(REPO_ROOT);
+
 const chmodAsync = promisify(chmod);
 const unlinkAsync = promisify(unlink);
 const writeFileAsync = promisify(writeFile);
 
-const PRECOMMIT_GIT_HOOK_SCRIPT_PATH = resolve(REPO_ROOT, '.git/hooks/pre-commit');
+async function getPrecommitGitHookScriptPath(rootPath) {
+  return new Promise((pResolve, pReject) => {
```

Comment:
How about preserving the verbs in these function names?

```suggestion
  return new Promise((resolvePromise, rejectPromise) => {
```